### ISSUE TITLE
For which inputs are gradients computed?

### ISSUE THREAD

**Issue by arunmallya** [tensorflow/tensorflow#6025]
What is a good way to determine which inputs of a tensorflow op are backpropagated through?
For example, consider the [crop_and_resize](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/g3doc/api_docs/python/functions_and_classes/shard2/tf.image.crop_and_resize.md) op. The description sounds rather straightforward - 
>"Extracts crops from the input image tensor and bilinearly resizes them". 

However, if you dig through the code, this function pops up [_CropAndResizeGrad](https://github.com/tensorflow/tensorflow/blob/33e88f6d7b69f796d4fad452394237448f62b976/tensorflow/python/ops/image_grad.py#L77), which states 

> "We back-propagate to the image only when the input image tensor has floating point dtype but we always back-propagate to the input boxes tensor.",

which indicates that the input bounding box coordinates will get gradients.

It might just be me, but I have a hard time telling which input tensor variables will get gradients and which will not. Is there a standard way? Should this be part of the documentation? 

Thanks.

**Comment by martinwicke** [comment]
It should be part of the documentation. Ideally, every op that does *not* have gradients for some of its inputs should call this out. Adding those to the docs when you see them would be a service to us all.

**Comment by jrock08** [comment]
While in an ideal world, each op that doesn't have gradients for some of it's inputs would be enough, it's tough to verify.  Shouldn't this be the sort of thing that is documented by default for all functions as a requirement for merge.  Rather than leaving this as something that is documented when there isn't a gradient for some input which is easy to overlook at code review.  An alternative is requiring that each argument in the Arg: portion of the documentation explicitly states whether gradients are passed.  This would be easy to review and also serve to highlight which functions haven't been fixed yet.

**Comment by tfboyd** [comment]
Automatically closing due to lack of recent activity. We hope that you were able to resolve it on your own. However, since this is a support issue rather than a bug or feature request, you will probably get more information by posting it on StackOverflow.
